pkgs/development/python-modules/pip-chill/default.nix

```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildPythonPackage,
+  setuptools,
+  pip,
+  pythonAtLeast,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "pip-chill";
+  version = "1.0.3";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "rbanffy";
+    repo = "pip-chill";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-oWq3UWBL5nsCBUkaElashZKvm7pN3StJNubgU++8YFs=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = lib.optionals (pythonAtLeast "3.12") [ setuptools ];
+
+  nativeCheckInputs = [
+    pip
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "pip_chill" ];
```

Comment:
The order of parameter mostly follow the [phases order](https://nixos.org/manual/nixpkgs/stable/#sec-stdenv-phases). Which is the order of execution.

I would add `pythonImportsCheck` to `installCheck` phase. Meaning, just before meta, in this case.